src/widgets/Scrollbar.hpp

```diff
@@ -1,53 +1,130 @@
+/// be at after the animation. The latter is used for example to check if the
+/// scrollbar is at the bottom.
+///
+/// `minimum` and `maximum` are used to map scrollbar positions to
+/// (message-)buffer indices. The buffer is of size `maximum - minimum` and an
+/// index is computed by `scrollbarPos - minimum` - thus a scrollbar position
+/// of a message is at `index + minimum.
+///
+/// @cond src-only
+///
+/// The following illustrates a scrollbar in a channel view with seven
+/// messages. The scrollbar is at the bottom. No animation is active, thus
+/// `currentValue = desiredValue`.
+///
+///                ┌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┐←╌╌╌ minimum
+///                  Alice: This message is quite           = 0
+///             ┬  ╭─────────────────────────────────╮←╮
+///             │  │ long, so it gets wrapped        │ ┆
+///             │  ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤ ╰╌╌╌ currentValue
+///             │  │ Bob: are you sure?              │       = 0.5
+///             │  ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤       = desiredValue
+/// pageSize ╌╌╌┤  │ Alice: Works for me... try for  │       = maximum
+///  = 6.5      │  │ yourself                        │         - pageSize
+///             │  ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤       = bottom
+///             │  │ Bob: I'm trying to get my really│       ⇒ atBottom = true
+///             │  │ long message to wrap so I can   │
+///             │  │ debug this issue I'm facing...  │
+///             │  ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+///             │  │ Bob: Omg it worked              │
+///             │  ├╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+///             │  │ Alice: That's amazing!         ╭┤ ┬
+///             │  │╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌││ ├╌╌ thumbRect.height()
+///             │  │ Bob: you're right              ╰┤ ┴
+///             ┴╭→╰─────────────────────────────────╯
+///              ┆
+///           maximum
+///            = 7
+/// @endcond
+///
+/// When messages are added at the bottom, both maximum and minimum are offset
+/// by 1 and after a layout, the desired value is updated, causing the content
+/// to move. Afterwards, the bounds are reset (potentially waiting for the
+/// animation to finish).
+///
+/// While scrolling is paused, the desired (and current) value won't be
+/// updated. However, messages can still come in and "shift" the values in the
+/// backing ring-buffer. If the current value would be used, the messages would
+/// still shift upwards (just at a different offset). To avoid this, there's a
+/// _relative current value_, which is `currentValue - minimum`. It's the
+/// actual index of the top message in the buffer. Since the minimum is shifted
+/// by 1 when messages come in, the view will remain idle (visually).
 class Scrollbar : public BaseWidget
 {
     Q_OBJECT
 
 public:
-    Scrollbar(size_t messagesLimit, ChannelView *parent = nullptr);
```

Comment:
Could use [`gsl::not_null`](https://github.com/microsoft/GSL/blob/main/docs/headers.md#user-content-H-pointers-not_null).

Reply:
the parent could technically be changed to a QObject since we don't use it as a ChannelView, nullptr here is currently used in tests